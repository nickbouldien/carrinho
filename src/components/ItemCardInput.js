import React, { Component } from 'react';
import { Form, Input } from 'semantic-ui-react';

class ItemCardInput extends Component {
  state = {
    amt: this.props.item.amt || 1,
    initialAmt: this.props.item.amt || 0,
  };

  handleChange = ({ target: { value } }) => {
    this.setState({
      amt: parseInt(value, 10),
    });
  };

  render() {
    console.log("state: ", this.state);
    console.log("props: ", this.props);

    const { amt, initialAmt } = this.state;
    const { item, location, onClickFn } = this.props;
    return (
      <Form onSubmit={ () => onClickFn(item, amt, initialAmt) }>
        <Form.Field>
          <label htmlFor={`item-${ item.id }-quantity`}>
            Amount
          </label>
          <Input
            id={`item-${ item.id }-quantity`}
            min={0}
            max={item.stock}
            name={`quantity`}
            onChange={this.handleChange}
            type='number'
            value={amt}
          />
          <Form.Button
            disabled={ amt === 0 }
            content={ location === "cart" ? "Remove From Cart" : "Add to cart" }
          />
        </Form.Field>
      </Form>
    );
  }
}

export default ItemCardInput;
