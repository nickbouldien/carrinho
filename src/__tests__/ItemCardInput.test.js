import React from 'react';
import ReactDOM from 'react-dom';

import ItemCardInput from '../components/ItemCardInput';

describe("tests for ItemCardInput component", () => {
  const mockFn = jest.fn();
  const mockItem = {
    amt: 12,
    id: "abc123",
    image: "",
    name: "thing",
    stock: 234,
  };

  const container = document.createElement('div');
  ReactDOM.render(
    <ItemCardInput onClickFn={mockFn} item={mockItem} />,
    container
  );

  const form = container.querySelector("form");
  const { quantity } = form.elements;

  test("Button calls mockFn when clicked", () => {
    expect(parseInt(quantity.value, 10)).toBe(mockItem.amt);

    form.dispatchEvent(new window.Event('submit'));
    expect(mockFn).toHaveBeenCalledTimes(1);
    expect(mockFn).toHaveBeenCalledWith(
      mockItem,
      mockItem.amt,
      mockItem.amt
    );
  });

  test("mockFn gets called when button is clicked", () => {
    const button = container.querySelectorAll('button');
    expect(button).toHaveLength(1);

  });

  test.skip("Button renders the correct verbiage based on current location", () => {
    const button = container.querySelectorAll('button');
    console.log(button);
    expect(button.textContent).toMatch("asdfasf");
  });

});
