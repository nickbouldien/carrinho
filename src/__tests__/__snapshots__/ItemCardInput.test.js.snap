// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for ItemCardInput component snapshot test 1`] = `
<form
  class="ui form"
>
  <div
    class="field"
  >
    <label
      for="item-abc123-quantity"
    >
      Amount
    </label>
    <div
      class="ui input"
    >
      <input
        type="number"
        min="0"
        max="234"
        id="item-abc123-quantity"
        name="quantity"
        value="12"
      />
    </div>
    <div
      class="field"
    >
      <button
        class="ui button"
        role="button"
      >
        Remove From Cart
      </button>
    </div>
  </div>
</form>
`;
